code/modules/mob/living/simple_animal/bot/construction.dm

```diff
@@ -347,6 +348,47 @@
 					S.name = created_name
 					qdel(src)
 
+
+//Honkbot Assembly
+
+/obj/item/honkbot_assembly
+	name = "incomplete honkbot assembly"
+	desc = "The clown's up to no good once more"
+	icon = 'icons/mob/aibots.dmi'
+	icon_state = "honkbot_arm"
+	var/build_step = ASSEMBLY_FIRST_STEP
+	var/created_name = "honkbot"
+
+/obj/item/honkbot_assembly/attackby(obj/item/I, mob/user, params)
+
+	if(isprox(I) && isbuild_step(ASSEMBLY_FIRST_STEP))
+		if(!user.temporarilyRemoveItemFromInventory(I))
+			return
+		build_step++
+		to_chat(user, "<span class='notice'>You add the prox sensor to [src]!</span>")
+		icon_state = "honkbot_proxy"
+		name = "Incomplete Honkbot Assembly"
+		qdel(I)
+
+	else if(istype(I, /obj/item/bikehorn) && isbuild_step(ASSEMBLY_SECOND_STEP))
+		if(!user.temporarilyRemoveItemFromInventory(I))
+			return
+		to_chat(user, "<span class='notice'>You add the bike horn to [src]! Honk!</span>")
+		var/mob/living/simple_animal/bot/honkbot/S = new(drop_location())
```

Comment:
do S.loc since this is a transofrmation

Reply:
Was I meant to replace this line, or simply add S.loc after?

Reply:
new(S.loc) will produce an self-referencing infinite loop. 
I can either use new(loc), or use S.loc after generating the item.